certs-manager/src/main/java/io/strimzi/certs/SecretCertProvider.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Provides Kubernetes secrets containing certificates
+ */
+public class SecretCertProvider {
+
+    private static final String DEFAULT_KEY_KEY = "tls.key";
+    private static final String DEFAULT_KEY_CERT = "tls.crt";
+
+
+    /**
+     * Create a Kubernetes secret containing the provided private key and related certificate
+     * using default values for the keys in the Secret data section
+     *
+     * @param name Secret name
+     * @param keyFile private key to store
+     * @param certFile certificate to store
+     * @return the Secret
+     * @throws IOException
+     */
+    public Secret createSecret(String name, File keyFile, File certFile) throws IOException {
```

Comment:
I wonder if it would make sense to encapsulate the `File` objects into something like `TLSKey`, `TLSCert` etc. to have some kind of type control over what we pass where.